doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+equations.
+
+What's wrong with solve():
+--------------------------
+
+SymPy already has a pretty powerful ``solve`` function. But it has a lot of major
+issues
+
+1. It doesn't have a consistent output for various types of solutions
+   It needs to return a lot of types of solutions consistently:
+
+   * single solution : `x == 1`
+   * Multiple solutions: `x^2 == 1`
+   * No Solution: `x^2 + 1 == 0` ; x is real
+   * Interval of solution: `floor(x) == 0`
+   * Infinitely many solutions: `sin(x) == 0`
+   * Multivariate functions with point solutions: `x^2 + y^2 == 0`
+   * Multivariate functions with non point solution: `x^2 + y^2 == 1`
+   * System of equations: `x + y == 1` and `x - y == 0`
+   * Relational: `x > 0`
+   * And the most important case "We don't Know"
+
+2. The input API is also a mess, there are a lot of parameters. Many of them
+   are not needed and they make it hard for the user and the developers to
+   work on solvers.
+
+3. There are cases like finding the maxima and minima of function using
+   critical points where it is important to know if it has returned all the
+   solutions. ``solve`` does not guarantee this.
+
+
+Why Solveset?
+-------------
+
+* ``solveset`` has a cleaner input and output interface: ``solveset`` returns
+  a set object and a set object takes care of all types of output. For
+  cases where it doesn't "know" all the solutions a `ConditionSet` with partial
+  solution is returned. For input it only takes the equation, the variables
+  to solve for & the optional argument `domain` in which the equations has to
+  be solved.
+
+* ``solveset`` can return infinitely many solutions. For example solving for
+  `sin(x) = 0` returns `\{2 n \pi \| n \in \mathbb{Z}\} \cup \{2 n \pi + \pi \| n \in \mathbb{Z}\}`, whereas `solve`
+  only returns [0, π].
+
+* There is a clear code level and interface level separation between solvers
+  for equations in complex domain and equations in real domain. For example
+  solving `exp(x) == 1` when `x` is to be solved in complex `domain`, returns
+  the set of all solutions that is `\{2 n i \pi \| n \in \mathbb{Z}\}`, whereas if `x` is to be
+  solved in the Real `domain` then only `\{0\}` is returned.
+
+
+Why do we use Sets as an output type?
+-------------------------------------
+
+SymPy has a well developed sets module, which can represent most of the set
+containers in Mathematics such as:
+
+ * ``FiniteSet``
+   Represents a finite set of discrete numbers.
+
+ * ``Interval`` 
+   Represents a real interval as a Set.
+
+ * ``ProductSet``
+   Represents a Cartesian Product of Sets.
+
+ * ``ImageSet``
+   Represents the Image of a set under a mathematical function
+
+    >>> from sympy import ImageSet, S, Lambda
+    >>> from sympy.abc import x
+    >>> from sympy.sets import ImageSet
+    >>> squares = ImageSet(Lambda(x, x**2), S.Naturals)  # {x**2 for x in N}
+    >>> 4 in squares
+    True
+
+ * ``ComplexPlane``
+   Represents the Set of all Complex Numbers
```

Comment:
But the complex plane is the *whole* complex plane, whereas an instance of `ComplexPlane` is generally not.